src/app/resources.py

```diff
@@ -77,198 +86,197 @@ def delete(self, *args, **kwargs):
+                session.query(User).filter(User.id == user_id).delete()
+                session.commit()
+                return f'User ID:{user_id} has been DELETED.', 200  # OK
+            else:
+                return f'User ID {user_id} - Not Found', 404  # Not Found
+        except SQLAlchemyError as err:
+            return str(err), 500  # Internal Server Error
 
-        current_user = User.filter_by_email(args["email"], session)
-        if current_user and current_user.compare_hash(args["password"]):
-            sess["email"] = args["email"]
-            sess.permanent = True
-            app.permanent_session_lifetime = datetime.timedelta(minutes=60)
-            return "Logged in as {}".format(current_user.email)
 
-        return "Could not verify your login!", 401, {"WWW-Authenticate": 'Basic realm="Login Required"'}
+@api.representation('/json')
+class PasswordListResource(EntityListResource):
+    @api.expect(password_post)
+    def post(self, user_id):
+        json_data = request.get_json()
+        if not json_data or not isinstance(json_data, dict):
+            return 'No input data provided', 400  # Bad Request
 
+        # Validate and deserialize input
+        try:
+            data = PasswordSchema().load(json_data)
+        except ValidationError as err:
+            return str(err), 422  # Unprocessable Entity
 
-@api.representation('/json')
-class Logout(Resource):
-    """
-    Logout resource
+        if not session.query(User).filter(User.id == user_id).first():
+            return f'User ID {user_id} - Not Found', 404  # Not Found
 
-    Remove the username from the session
-    """
+        # crate a new password
+        try:
+            session.add(Password(user_id, data))
+            session.commit()
+            return 'PASSWORD ADDED', 200  # OK
+        except SQLAlchemyError as err:
+            return str(err), 500  # Internal Server Error
 
-    def get(self):
-        sess.pop("email", None)
-        return "Dropped!"
+    def get(self, user_id):
+        try:
+            if not session.query(User).filter(User.id == user_id).first():
+                return f'User ID {user_id} - Not Found', 404  # Not Found
+            passwords = session.query(Password).filter(Password.user_id == user_id).all()
+            passwords_serialized = []
+            for password in passwords:
+                passwords_serialized.append(password.serialize)
+            return {'all user passwords': passwords_serialized}, 200  # OK
+        except SQLAlchemyError as err:
+            return str(err), 500  # Internal Server Error
+
+
+@api.representation('/json')
+class PasswordResource(EntityResource):
+    def get(self, user_id, pass_id):
+        try:
+            if not session.query(User).filter(User.id == user_id).first():
+                return f'User ID {user_id} - Not Found', 404  # Not Found
+            password = session.query(Password) \
```

Comment:
As Vanya said, if you use your code more than 1 time, you have to create a function for that like on lines 253 and 269

Reply:
Add methods into Password model as we have it in User model